File: 2020/src/day0.jl
module Day0

include("InputHelper.jl")

function get_input()::Array{String,1}
    InputHelper.parse(line -> line, "0", "\n")
end

function solve()

    input = get_input()

    solve_part_one(input), solve_part_two(input)
end

function solve_part_one(input)
    nothing
end


function solve_part_two(input)
    nothing
end

end

File: aoc-2022/src/day0.rs
use crate::parser::parser;

type Input = Vec<i32>;

pub fn solve() {
    let input = parse();
    let one = solve1(input.to_vec());
    println!("Part one: {}", one);
    let two = solve2(input.to_vec());
    println!("Part two: {}", two);
}

fn parse() -> Input {
    parser::parse(0, "\n", |line| line.parse::<i32>().unwrap())
}

fn solve1(_input: Input) -> i32 {
    println!("Part one: {}", -1);
    -1
}

fn solve2(_input: Input) -> i32 {
    println!("Part two: {}", -1);
    -1
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn part_one_test() {
        let input = parse();
        assert_eq!(solve1(input), -1);
    }

    #[test]
    fn part_two_test() {
        let input = parse();
        assert_eq!(solve2(input), -1);
    }
}
